internal/cli/finalizer.go

```diff
@@ -0,0 +1,54 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+const timeout = 5
+
+type Finalizer struct {
+	notify func(c chan<- os.Signal, sig ...os.Signal)
+	exit   func(int)
+	funcs  []func()
+}
+
+func NewFinalizer() *Finalizer {
+	return &Finalizer{
+		notify: signal.Notify,
+		exit:   os.Exit,
+	}
+}
+
+func (f *Finalizer) Add(function func()) {
+	f.funcs = append(f.funcs, function)
+}
+
+func (f *Finalizer) SetupCloseHandler() {
```

Comment:
Why have a separate function to activate the finaliser? I would expect this code to be part of `NewFinalizer()`.
The constructor should deliver a ready to use object

Reply:
It is ready to use ;) And you can use it by the `SetupCloseHandler` method. In my opinion, both solutions are acceptable.

Reply:
If I call `NewFinalizer()`, I always have to call `SetupCloseHandler()` afterwards, there is no use case without calling the second function.
So we might as well simplify the contract and need only to call the constructor.

Reply:
But this will complicate my tests. After this PR I will take another issue around the finalizer (Issue is not created yet). Can we talk about it during next review?